### PR TITLE
Clarify rolling-update process (#13094)

### PR DESCRIPTION
**Pull request by sunsingerus**
Introduce proper values for deployment rolling-update parameters
1. MaxUnavailable
2. MaxSurge

### CODE REVIEW

**Comment by sunsingerus** [comment]
Explanation on what is done and why is available [in issue](https://github.com/mattermost/mattermost-server/issues/13094#issuecomment-555004029)

**Comment by gabrieljackson** [comment]
Thanks for this @sunsingerus! We will be reviewing this soon.

File: cmd/manager/main.go [hunk 1]
```diff
@@ -4,13 +4,13 @@ import (
...
 	"flag"
 	"fmt"
-	"os"
-	"runtime"
-
 	"github.com/mattermost/mattermost-operator/pkg/apis"
 	"github.com/mattermost/mattermost-operator/pkg/controller"
 	"github.com/mattermost/mattermost-operator/pkg/log"
 	"github.com/mattermost/mattermost-operator/version"
+	"os"
```
Comment (gabrieljackson) [inline cmd/manager/main.go]: nit: these three imports make a bit more sense near the top where they originally were.

File: cmd/manager/main.go [hunk 2]
```diff
@@ -26,8 +26,12 @@ import (
 
 // Change below variables to serve metrics on different host or port.
 var (
-	metricsHost       = "0.0.0.0"
+	// metricsHost specifies host to bind to for serving prometheus metrics
+	metricsHost = "0.0.0.0"
+	// metricsPort specifies port to bind to for serving prometheus metrics
 	metricsPort int32 = 8383
+	// syncPeriod specified period to reconcile resources in order to reduce spread between local cache and k8s
```
Comment (gabrieljackson) [inline cmd/manager/main.go]: nit:
```suggestion
	// syncPeriod specifies the period to reconcile resources in order to reduce spread between local cache and k8s
```

File: cmd/manager/main.go (hunk 1 above)
Comment (sunsingerus) [reply cmd/manager/main.go]: Yes, agreed, my IDE has its own opinion - tries to order alphabetically all imports

**Comment by gabrieljackson** [comment]
@sunsingerus, this looks great! As soon as you can address the failing test we can get this merged:

https://circleci.com/gh/mattermost/mattermost-operator/3311?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

You may want to ensure your branch has all the code from master if you haven't done that recently. Let us know if you become blocked on anything.

**Comment by sunsingerus** [comment]
Yes, sure I'll take a look on failed test. Not that much of a change introduced. Will return back soon

**Comment by sunsingerus** [comment]
By somewhat unclear reason tests are sometimes unable to launch mysql Pod and are failed on `waitForStatefulSet()` Most interesting, Pod is not starting in different tests

**Comment by gabrieljackson** [comment]
Thanks for the update @sunsingerus. I was just about to write you to let you know that we are noticing the same failure on master when nothing seems to have caused this regression lately.

At this point, let's leave this PR open and the cloud team will investigate the failure. We will let you know when it is taken care of.

**Comment by gabrieljackson** [comment]
Based on the tests passing, it appears that we are right on the edge of the resources available on the e2e test cluster. We will merge this and continue to investigate the test failures.

Thanks @sunsingerus!

**Pull request merged**